ADR/00xx-component-references.md

```diff
@@ -0,0 +1,147 @@
+* [build-service] will propagate digest references as a PR to the bundle image
+* [integration-service] will post followup checks on the bundle PR saying “don’t merge this” until all operand PRs are merged.
+
+Think about branches:
+
+* User team updates their operand images with branch feature-1234 (or whatever).
+* [build-service] uses that same branch in its PR to the bundle repo.
+* If user submits three operand image PRs and uses the same branch name for all – then the digests all pile up on the same bundle branch, the same bundle PR.
+* If the user submits three operand image PRs and uses different branch names for all – then the digests are split among three different bundle branches, three different bundle PRs.
+
+---
+
+### OLM Operators, with components in the same repo
+
+Another OLM Operator Scenario: an operator git repo contains both the controller code and the bundle metadata.
+
+* The user has two Components, that both point to the same repo.
+* The bundle Component declares that it embeds references to the controller image Component, by way of a new `ComponentReference` CR.
+* [integration-service] will always skip testing for the controller component because it is a *referent* Component.
+* [build-service] will propagate the digest reference as a PR to the bundle image, which happens to be the same git repository as the controller Component.
+  * The user submitted their controller image update on the `new-feature` branch.
+  * [build-service] could do one of the following:
+    * ❌ Push its commits with the new image digest references to the same feature branch.
+      * Will [build-service] -- acting as the TAP github App -- have rights to push to that branch? Probably not.
+    * 🤔 Submit its commits as a suggestion on the original PR, which the user can accept or not.
+      * This would be “odd” for build service, but maybe nice from the user’s point of view. Just one PR to work with.
+    * ❌ Push its commits with the new image digest references to a new feature branch based on the original feature branch, called `new-feature-build-service`
+      * This could be nice, but the PR will trigger new builds of the original image since it contains code changes for the controller (here, I am assuming that we can limit builds in PaC based on contextDirectories for multi-Component repos; we don’t have this today, but assume that we do).
+    * ✅ Push its commits with the new image digest references to a new feature branch that is not based on the original feature branch, called `new-feature-from-build-service`
+      * This works like it does in most other cases. Let the user merge the original PR, and only after that will [integration-service] update the checks for `new-feature-from-build-service` to say “okay to merge now!”
+
+## Open Questions
+
+* To give the user some control, should we make [build-service] respect [renovatebot
+  configuration files](https://docs.renovatebot.com/configuration-options/)? Does that still make
```

Comment:
Is it possible to respect all but some subset of renovatebot configurations? If this functionality is enabled, it wouldn't make sense to limit the frequency of updates, for example.

Reply:
Or, would respecting some-but-not-all configuration be even more confusing for users?

Reply:
I don't know how the configuration of renovatebot might look. Would it be possible to separate the configuration for AppStudio as a whole vs. ComponentReferences specifically? If it is possible for users to customize it then they might want to in order to enable quick propagation of these changes while reducing overall PR volume from build pipeline updates.

Reply:
Yeah, it exists as a file in the user's repo: https://docs.renovatebot.com/configuration-options/

We could ask the user to put one file in place for "normal" renovatebot control, and a second file with a different name but the same schema (or a subset of the normal schema) to control ComponentReference behavior. Say, `.renovaterc.json` and `.references.json`.

Reply:
But - I don't have a concrete thing yet that I can imagine the user would want to control. We should analyze the options and introduce this only if we know of a thing a user would actually want to use it for.

Reply:
I think I am agreeing with where you are at -- the default case would be to not respect the configurations. If requested, a separate configuration file can be respected.